File: backend/src/config/cron.js
import cron from "cron";
import https from "https";

const job = new cron.CronJob("*/14 * * * *", function () {
  https
    .get(process.env.API_URL, (res) => {
      if (res.statusCode === 2000) console.log("Làm mới dữ liệu thành công");
      else
        console.log(
          "Làm mới dữ liệu thất bại với mã trạng thái: ",
          res.statusCode
        );
    })
    .on("error", (err) => {
      console.log("Lỗi khi làm mới dữ liệu: ", err.message);
    });
});

export default job;

File: backend/src/config/cron.py
import os
import httpx 
from apscheduler.schedulers.background import BackgroundScheduler

API_URL = os.getenv("API_URL")

scheduler = BackgroundScheduler()

def refresh_data():
    try:
        res = httpx.get(API_URL, timeout=10)

        if res.status_code == 200:
            print("Làm mới dữ liệu thành công")
        else:
            print(f"Làm mới thất bại: {res.status_code}")

    except Exception as e:
        print("Lỗi khi làm mới dữ liệu:", str(e))


def start_cron():
    scheduler.add_job(refresh_data, "interval", minutes=14)
    scheduler.start()